packages/strapi-database/lib/migration-manager.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+const debug = require('debug')('strapi-database:migration');
+const { isFunction } = require('lodash/fp');
+
+class MigrationManager {
+  constructor(db) {
+    debug('Initialize migration manager');
+    this.db = db;
+    this.migrations = [];
+  }
+
+  register(migration) {
+    debug('Register migration');
+    this.migrations.push(migration);
+  }
+
+  async runMigration(fn, options, context = {}) {
+    debug('Run migration');
+    await this.runBefore(options, context);
+    await await fn(options, context);
+    await this.runAfter(options, context);
+  }
+
+  async runBefore(options, context) {
+    debug('Run before migrations');
+
+    for (const migration of this.migrations) {
+      if (isFunction(migration.before)) {
+        await migration.before(options, context);
+      }
+    }
+  }
+
+  async runAfter(options, context) {
+    debug('Run after migrations');
+
+    for (const migration of this.migrations.slice(0).reverse()) {
```

Comment:
Why do you need to `slice(0)` here? What does it change from `this.migrations.reverse()`?

Reply:
reverse mutates the object :)

Reply:
Makes sense, didn't think about this :+1: